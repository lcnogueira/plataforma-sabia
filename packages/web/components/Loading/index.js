export { default as Spinner } from './Spinner';
export { default as Wrapper } from './Wrapper';
