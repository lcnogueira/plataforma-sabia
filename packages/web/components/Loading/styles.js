import styled from 'styled-components';
import BeatLoader from 'react-spinners/BeatLoader';

export const Container = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	transition: 1s height;
`;

export const Loader = styled(BeatLoader).attrs(({ theme: { colors } }) => ({
	size: 35,
	color: colors.primary,
}))``;
