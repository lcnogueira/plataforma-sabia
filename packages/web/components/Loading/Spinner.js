import React from 'react';
import PropTypes from 'prop-types';

import { Container, Loader } from './styles';

const Spinner = ({ variant }) => {
	return (
		<Container>
			<Loader variant={variant} />
		</Container>
	);
};

Spinner.propTypes = {
	variant: PropTypes.oneOf(['primary', 'secondary', 'white']),
};

Spinner.defaultProps = {
	variant: 'primary',
};

export default Spinner;
