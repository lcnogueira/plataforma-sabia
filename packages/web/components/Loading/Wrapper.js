import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import Spinner from './Spinner';
import { HiddenWrapper } from './styles';

const Wrapper = ({ loading, variant, children, alwaysRenderChildren }) => {
	const spinnerComponent = createElement(Spinner, { variant });

	if (alwaysRenderChildren) {
		return (
			<>
				{loading && spinnerComponent}
				<HiddenWrapper active={loading}>{children}</HiddenWrapper>
			</>
		);
	}

	return loading ? spinnerComponent : children;
};

Wrapper.propTypes = {
	loading: PropTypes.bool,
	children: PropTypes.node.isRequired,
	variant: PropTypes.oneOf(['primary', 'secondary', 'white']),
	alwaysRenderChildren: PropTypes.bool,
};

Wrapper.defaultProps = {
	loading: true,
	variant: 'primary',
	alwaysRenderChildren: false,
};

export default Wrapper;
