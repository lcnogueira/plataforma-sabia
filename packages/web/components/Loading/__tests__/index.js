import React from 'react';
import { render } from 'test-utils';
import Loading from '..';

describe('Loading component', () => {
	const childrenText = 'hi';

	it('render loading or children correctly', () => {
		const { container, rerender } = render(
			<Loading loading={false}>
				<h1>{childrenText}</h1>
			</Loading>,
		);

		expect(container.querySelector('h1').textContent).toEqual(childrenText);

		rerender(
			<Loading loading>
				<h1>{childrenText}</h1>
			</Loading>,
		);

		expect(container.querySelector('h1')?.textContent).toBeFalsy();
		expect(container).toMatchSnapshot();
	});

	it('render with another variants', () => {
		const { container, rerender } = render(
			<Loading variant="secondary">
				<h1>{childrenText}</h1>
			</Loading>,
		);

		rerender(
			<Loading variant="white">
				<h1>{childrenText}</h1>
			</Loading>,
		);

		expect(container).toMatchSnapshot();
	});
});
