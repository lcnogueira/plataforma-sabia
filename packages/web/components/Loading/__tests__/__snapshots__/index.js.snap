// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component render loading or children correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > div {
  padding: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-q6dswj"
      />
      <div
        class="css-1dcll36"
      />
      <div
        class="css-1l582lg"
      />
      <div
        class="css-phvgz1"
      />
      <div
        class="css-1un1iog"
      />
    </div>
  </div>
</div>
`;

exports[`Loading component render with another variants 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > div {
  padding: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-1uv12al"
      />
      <div
        class="css-a00mw7"
      />
      <div
        class="css-xp33ak"
      />
      <div
        class="css-11jye0"
      />
      <div
        class="css-1remiyp"
      />
    </div>
  </div>
</div>
`;
